webapp-django/crashstats/crashstats/management/commands/archivescraper.py

```diff
@@ -118,6 +118,25 @@ def __init__(self, base_url, num_workers, verbose):
         self.base_url = base_url
         self.num_workers = num_workers
         self.verbose = verbose
+        self.msgs = []
+
+    def worker_write(self, msg, verbose=False):
+        """Writes to stdout or msgs buffer.
+
+        Use this in workers rather than printing directly to stdout so it can
+        be passed to the parent process correctly.
+
+        :arg str msg: the message to write
+        :arg bool verbose: whether to write this only if in verbose mode
+
+        """
+        if verbose and not self.verbose:
+            return
+
+        if self.num_workers > 1:
+            self.msgs.append(msg)
+        else:
+            print(msg)
```

Comment:
If there's one worker, then everything is happening in the parent process and we can skip the whole message buffer thing. If there's more than one worker, then parts are running in subprocesses.

Now that I'm thinking about this, I think I may nix the short-cut here. I'd rather it was getting tested in the tests.